catalogd/internal/source/containers_image.go

```diff
@@ -37,6 +39,23 @@ type ContainersImageRegistry struct {
 }
 
 func (i *ContainersImageRegistry) Unpack(ctx context.Context, catalog *catalogdv1.ClusterCatalog) (*Result, error) {
+	res, err := i.unpack(ctx, catalog)
```

Comment:
Lines 59-75 (the first lines of `unpack` should be moved into `Unpack` to:
1. Get a common logger to use (we should get it from the context, not from the root `ctrl.Log`)
2. Only have to call i.SourceContextFunc(l) once


Then we could just pass the logger and the `SystemContext` to `unpack()`.

Reply:
I just wrapped the minimum to get the error... but yes, this is a better idea.